internal/driver/fetch.go

```diff
@@ -580,17 +581,35 @@ func adjustURL(source string, duration, timeout time.Duration) (string, time.Dur
 
 // httpGet is a wrapper around http.Get; it is defined as a variable
 // so it can be redefined during for testing.
-var httpGet = func(source string, timeout time.Duration) (*http.Response, error) {
+var httpGet = func(s *source, source string, timeout time.Duration) (*http.Response, error) {
+	var cert tls.Certificate
+
 	url, err := url.Parse(source)
 	if err != nil {
 		return nil, err
 	}
 
-	var tlsConfig *tls.Config
-	if url.Scheme == "https+insecure" {
-		tlsConfig = &tls.Config{
-			InsecureSkipVerify: true,
+	tlsConfig := &tls.Config{}
+	if s.HTTPSCert != "" && s.HTTPSKey != "" {
+		cert, err = tls.LoadX509KeyPair(s.HTTPSCert, s.HTTPSKey)
+		if err != nil {
+			return nil, err
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
+
+	if s.HTTPSCA != "" {
+		caCertPool := x509.NewCertPool()
+		caCert, err := ioutil.ReadFile(s.HTTPSCA)
+		if err != nil {
+			return nil, err
 		}
+		caCertPool.AppendCertsFromPEM(caCert)
+		tlsConfig.RootCAs = caCertPool
+	}
+
+	if url.Scheme == "https+insecure" {
```

Comment:
Should `internal/symbolizer/symbolizer.go` be updated as well?

Reply:
Do you mean the `postURL` in `symbolizer.go`?

Reply:
Yes.

Reply:
The problem is the `source` struct which has the command line flags is in `driver` pkg. That's why `Symbolizer()` takes `mode string` rather than `s *source`. In order to keep it clean I would need to create a new package for the `source` struct, do you have any preference where it should go?

Reply:
I don't think we should pass all of `source` struct to the symbolizer, it doesn't need most of that. Maybe something like this in `internal/plugin/plugin.go`:

```
type TLSParams struct {
  HTTPSCert, HTTPSKey, HTTPSCA string
}

and then pass it as to the symbolizer and to the fetcher as a new param.

Reply:
Changed the implementation to use `TLSParams`, thoughts?